python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -218,217 +229,167 @@ def _run(self, log, suppress_prompts, core_version):
     """
-    
+
     # possible update status states
-    (UP_TO_DATE,                    # all good, no update necessary  
-     UPDATE_POSSIBLE,              # more recent version exists
-     UPDATE_BLOCKED_BY_SG          # more recent version exists but SG version is too low.
-     ) = range(3)
+    (
+        UP_TO_DATE,                   # all good, no update necessary
+        UPDATE_POSSIBLE,              # more recent version exists
+        UPDATE_BLOCKED_BY_SG          # more recent version exists but SG version is too low.
+    ) = range(3)
 
-    def __init__(self, installation_root, logger, core_version=None):
+    def __init__(self, install_folder_root, logger, core_constraint_pattern=None):
         """
         Constructor
 
-        :param installation_root: The path to the installation to check. This is either a localized
-                                  Pipeline Configuration or a studio code location (omit the install folder).
-                                  Because we are passing this parameter in explicitly, the currently running
-                                  code base does not have to be related to the code base that is being updated,
-                                  e.g. you can run the updater as a totally separate thing.
+        :param install_folder_root: The path to the installation to check. This is either a localized
+                                   Pipeline Configuration or a studio code location (omit the install folder).
+                                   Because we are passing this parameter in explicitly, the currently running
+                                   code base does not have to be related to the code base that is being updated,
+                                   e.g. you can run the updater as a totally separate thing.
         :param logger: Logger to send output to.
         :param core_version: Version of the core to update to. If None, the core will be updated to the latest
-                             version.
+                             version. Defaults to None.
         """
         self._log = logger
 
-        (sg_app_store, script_user) = shotgun.create_sg_app_store_connection()
-        self._sg = sg_app_store
-        self._sg_script_user = script_user
+        from tank_vendor.shotgun_deploy import Descriptor, create_descriptor
 
-        self._local_sg = shotgun.get_sg_connection()
-        self._update_version = self.__get_core_version(core_version)
+        self._current_core_descriptor = create_descriptor(
+            shotgun.get_sg_connection(),
+            Descriptor.CORE,
+            {
+                "type": "app_store",
+                "version": pipelineconfig_utils.get_currently_running_api_version(),
+                "name": "core"
+            }
+        )
 
-        self._install_root = os.path.join(installation_root, "install")
+        self._new_core_descriptor = self._current_core_descriptor.find_latest_version(
```

Comment:
so with the current logic (now that I understand it more!), if someone has got a manual or dev descriptor, they would now get a different behaviour with the `tank core` command. I am not sure that's right. I think we shouldn't base the check on the latest version of the current descriptor, but the latest/explicit sought after version in the app store.